src/commands.rs

```diff
@@ -1215,23 +1233,55 @@ fn cmd_config(
     subcommand: &ConfigSubcommand,
 ) -> Result<(), CommandError> {
     ui.request_pager();
+    let settings = command.settings();
     match subcommand {
         ConfigSubcommand::List { name } => {
             let raw_values = match name {
-                Some(name) => command
-                    .settings()
-                    .config()
-                    .get::<config::Value>(name)
-                    .map_err(|e| match e {
-                        config::ConfigError::NotFound { .. } => {
-                            user_error("key not found in config")
-                        }
-                        _ => e.into(),
-                    })?,
-                None => command.settings().config().collect()?.into(),
+                Some(name) => {
+                    settings
+                        .config()
+                        .get::<config::Value>(name)
+                        .map_err(|e| match e {
+                            config::ConfigError::NotFound { .. } => {
+                                user_error("key not found in config")
+                            }
+                            _ => e.into(),
+                        })?
+                }
+                None => settings.config().collect()?.into(),
             };
             write_config_entry(ui, name.as_deref().unwrap_or(""), raw_values)?;
         }
+        ConfigSubcommand::Edit { config_args } => {
+            let edit_path = match (config_args.user, config_args.repo) {
+                // TODO(#531): Special-case for editors that can't handle viewing directories?
+                (true, false) => config_path()?.expect("No repo config path found to edit"),
```

Comment:
`.expect()` results in a panic if we fail to read the config. Since that is a normal case, we should return an error instead.